### PR TITLE
DROOLS-2820: [DMN Designer] [IE11] Node deletion throws an Error

### PR DESCRIPTION
**Pull request by danielzhe**

### CODE REVIEW

**Comment by manstis** [comment]
@danielezonca FYI,

DMN related PRs need @jomarko (from QE) to review too.

Stunner related PRs need @hasys (from QE too) to review.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/decision/tree/DecisionNavigatorTreeViewTest.java [hunk 1]
```diff
@@ -268,12 +268,15 @@ public void testRemove() {
...
+        final Element parentElement = mock(Element.class);
+        element.parentNode = parentElement;
 
         doReturn(element).when(treeView).findItem(item);
 
         treeView.remove(item);
 
-        verify(element).remove();
+        verify(parentElement).removeChild(element);
+        verify(element, never()).remove();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/decision/tree/DecisionNavigatorTreeViewTest.java]: Thanks for the test, please add also a negative one where we will check case when the `node.parenEleement` is null.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/decision/tree/DecisionNavigatorTreeView.java [hunk 2]
```diff
@@ -125,8 +125,11 @@ public boolean hasItem(final DecisionNavigatorItem item) {
 
     @Override
     public void remove(final DecisionNavigatorItem item) {
-        final Optional<Element> optionalElement = Optional.ofNullable(findItem(item));
-        optionalElement.ifPresent(Element::remove);
+        ofNullable(findItem(item)).ifPresent(element -> {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/decision/tree/DecisionNavigatorTreeView.java]: Could you please revisit whole `DecisionNavigatorTreeView` class and unify usage of `ofNullable` vs `Optional.ofNullable`. Currently class uses both. I think we should unify to use just one of them.

**Comment by jomarko** [comment]
@danielzhe manual check passed, so please have a look on my comments to code and I can approve, thanks for the fix!

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/decision/tree/DecisionNavigatorTreeView.java (hunk 2 above)
Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/decision/tree/DecisionNavigatorTreeView.java]: Good point. I've found just `ofNullable` more clean so I'll unify to this.

**Comment by danielzhe** [comment]
Jenkins please retest this.

**Comment by danielzhe** [comment]
@manstis merge, please.

**Comment by manstis** [comment]
@danielezonca Done. Thanks.

Normally, when there are multiple commits, I _squash_ them (when merging) into a single commit (as I have indeed done on this PR). If you wish commits to be preserved please comment the PR asking them not to be squashed when merged (it's occasionally useful, but on the whole it's best to squash).

**Comment by danielzhe** [comment]
@manstis Thank you for the info. P.S.: You've been marking "danielezonca", but I guess it was for me, DanielZhe :)

**Comment by manstis** [comment]
@danielzhe No freaking way! I'm sure I changed it.. but I can see I did not.

@danielezonca Sorry (again).

**Pull request merged**